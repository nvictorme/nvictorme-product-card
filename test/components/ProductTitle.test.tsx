import * as React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src';
import { Product } from '../../src/interfaces/interfaces';

const product: Product = {
  id: '1',
  title: 'Coffee Mug - Card',
};

describe('it', () => {
  it('renders ProductTitle without crashing', () => {
    const wrapper = renderer.create(<ProductTitle title={product.title} />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('renders ProductCard withoout crashing', () => {
    const wrapper = renderer.create(
      <ProductCard product={product} initialValues={{}}>
        {() => <ProductTitle />}
      </ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
