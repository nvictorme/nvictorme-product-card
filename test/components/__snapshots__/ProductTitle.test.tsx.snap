// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders ProductCard withoout crashing 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Card
  </span>
</div>
`;

exports[`it renders ProductTitle without crashing 1`] = `
<span
  className="productDescription undefined"
>
  Coffee Mug - Card
</span>
`;
